javascript_nodejs/50.contoso-cafe-bot/bot.js

```diff
@@ -0,0 +1,147 @@
+                // Do we have any oustanding dialogs? if so, continue them and get results
+                // No active dialog? start a new main dialog
+                await this.continueOrBeginMainDialog(context);
+                break;
+            }
+            case ActivityTypes.ConversationUpdate: {
+                
+                // Send a welcome card to any user that joins the conversation.
+                if(context.activity.membersAdded[0].name !== 'Bot') await this.welcomeUser(context);
+                break;
+            }
+            default: {
+                
+                // Handle other acivity types as needed.
+                break;            
+            }
+        }
+    }
+    /**
+     * Async method to continue or begin main dialog
+     * 
+     * @param {Object} context conversation context object
+     * 
+     */
+    async continueOrBeginMainDialog(context) {
+        // Create dialog context.
+        let dc = await this.dialogs.createContext(context);
+        
+        // Continue outstanding dialogs. 
+        let result = await dc.continue();
+        
+        // If no oustanding dialogs, begin main dialog
+        if(result.status === DialogTurnStatus.empty) {
+            await dc.begin(MainDialog.Name);
+        }
+    }
+    /**
+     * Async helper method to get on turn properties from cards or NLU using https://LUIS.ai
+     * 
+     * @param {Object} context conversation context object
+     * 
+     */
+    async getNewOnTurnProperties (context) {
+        // Handle card input (if any), update state and return.
+        if(context.activity.value !== undefined) return onTurnProperty.fromCardInput(context.activity.value);
+        
+        // Acknowledge attachments from user. 
+        if(context.activity.attachments && context.activity.attachments.length !== 0) {
+            await context.sendActivity(`Thanks for sending me that attachment. I'm still learning to process attachments.`);
+            return undefined;
+        }
+
+        // Nothing to do for this turn if there is no text specified.
+        if(context.activity.text === undefined || context.activity.text.trim() === '') return;
+
+        // make call to LUIS recognizer to get intent + entities
+        const LUISResults = await this.luisRecognizer.recognize(context);
+
+        // Return new instance of on turn property from LUIS results.
+        return onTurnProperty.fromLUISResults(LUISResults);
+    }
+    /**
+     * Async helper method to welcome the user.
+     * 
+     * @param {Object} context conversation context object
+     * 
+     */
+    async welcomeUser (context) {
+        // Welcome user.
+        await context.sendActivity(`Hello, I am the Contoso Cafe Bot!`);
+        await context.sendActivity(`I can help book a table, find cafe locations and more..`);
+        
+        // Send welcome card.
+        await context.sendActivity({ attachments: [CardFactory.adaptiveCard(welcomeCard)]});
```

Comment:
>{ attachments: [CardFactory.adaptiveCard(welcomeCard)]} [](start = 35, length = 55)

You should use `MessageFactory.attachment()` to send your card. You're not setting the activity `type` for instance.